### PR TITLE
Fixes a lot of runtimes caused when spiders are on spiderwebs

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

See title. Somebody used locate thinking it returned bool BUT IT DOESN'T and produced a lot of runtimes when spiders come into contact with spiderwebs as result.

## Changelog

:cl:
fix: Spiders no longer cause runtimes when coming into contact with spider webs. Also their AI no longer breaks due to this.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -86,7 +86,9 @@
 	return ..()
 
 /mob/living/simple_animal/hostile/giant_spider/mob_negates_gravity()
-	return (locate(/obj/structure/spider/stickyweb) in loc) || ..()
```
Comment (Time-Green) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:89]: wait what how does this not return a bool? 
"Hey is value x or y true?" 
"Spiderweb"

Comment (SmArtKar) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:89]: ...somehow. My guess would be that if ..() is false then || just returns the other one?

Comment (Time-Green) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:89]: wack

Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:89]: `x || y` resolves to x if its truthy, y if it's not

Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:89]: Bools in BYOND don't exist, this was returning a truthy value

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -86,7 +86,9 @@
...
+	if(locate(/obj/structure/spider/stickyweb) in loc)
+		return TRUE
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:91]:
```suggestion
	return ..()
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
This was also fixable by just adding `!isnull`

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:89]: This is a really useful behavior to know, and we use it a lot for things like default values, such as `var/name = input_name || "Other value"`

**Pull request merged**